opm/autodiff/MatrixBlock.hpp

```diff
@@ -319,6 +329,61 @@ namespace Detail
             }
         }
     }
+
+    //! calculates ret = A * B
+    template< class K>
+    static inline void multMatrix( const Dune::DynamicMatrix<K> &A,
+                                   const Dune::DynamicMatrix<K> &B,
+                                   Dune::DynamicMatrix<K> &ret )
+    {
+        typedef typename Dune::DynamicMatrix<K> :: size_type size_type;
+
+        const size_type m = A.rows();
+        const size_type n = A.cols();
+
+        assert(n == B.rows() );
+
+        const size_type p = B.cols();
+
+        ret.resize(m, p);
+
+        for( size_type i = 0; i < m; ++i )
```

Comment:
Cannot this function be implemented using `operator*` on DynamicMatrix?

Reply:
I am not totally sure. Most likely, they only support matrix-vector multiplication, which is typical for linear solver packages?